modules/dockerregistry/dockerregistry.go

```diff
@@ -0,0 +1,96 @@
+package dockerregistry
+
+import (
+	"context"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+// DockerRegistryContainer represents the DockerRegistry container type used in the module
+type DockerRegistryContainer struct {
+	testcontainers.Container
+}
+
+// RunContainer creates an instance of the DockerRegistry container type
+func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*DockerRegistryContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        "registry:2",
+		ExposedPorts: []string{"5000:5000/tcp"},
+		WaitingFor:   wait.ForExposedPort(),
+	}
+
+	genericContainerReq := testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	}
+
+	for _, opt := range opts {
+		opt.Customize(&genericContainerReq)
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
+	if err != nil {
+		return nil, err
+	}
+
+	return &DockerRegistryContainer{Container: container}, nil
+}
+
+// WithAuthentication customizer that will retrieve a htpasswd file into the htpasswd directory given in input
+func WithAuthentication(htpasswdPath string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		if req.Env == nil {
+			req.Env = make(map[string]string)
+		}
+
+		req.Env["REGISTRY_AUTH"] = "htpasswd"
+		req.Env["REGISTRY_AUTH_HTPASSWD_REALM"] = "Registry"
+		req.Env["REGISTRY_AUTH_HTPASSWD_PATH"] = "/auth/htpasswd"
+
+		if req.Mounts == nil {
+			req.Mounts = testcontainers.ContainerMounts{}
+		}
+
+		htpasswdMount := testcontainers.ContainerMount{
+			Source: testcontainers.GenericBindMountSource{
+				HostPath: htpasswdPath,
+			},
+			Target: "/auth",
+		}
+		req.Mounts = append(req.Mounts, htpasswdMount)
+	}
+
+}
+
+// WithData customizer that will retrieve a data directory and mount it inside the registry container
```

Comment:
Do we need a Docker mount for it? I wonder if copying the files/dir into the container, using the `req.Files` attribute would be something to consider.

@eddumelendez is there a preference when creating modules: using mounts or just copying files before the container start? 🤔

Reply:
The preference is to copy files.

Reply:
@davidesalerno then I'd advocate for converting all the Mounts into Files 🙏